src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs

```diff
@@ -76,17 +76,21 @@ internal static bool ValidateIssuerSigningKey(SecurityKey securityKey, SecurityT
 #if NET6_0_OR_GREATER
                 if (!string.IsNullOrEmpty(tokenIssuer) && !tokenIssuer.Contains(tenantIdFromToken, StringComparison.Ordinal))
                     throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidIssuerException(LogHelper.FormatInvariant(LogMessages.IDX40004, LogHelper.MarkAsNonPII(tokenIssuer), LogHelper.MarkAsNonPII(tenantIdFromToken))));
+
+                var v2TokenIssuer = openIdConnectConfiguration.Issuer?.Replace(AadIssuerValidator.TenantIdTemplate, tenantIdFromToken, StringComparison.Ordinal);
```

Comment:
Both of these replace calls allocate. Not sure if there is a way to avoid it yet.

Reply:
I don't see a way to avoid it today. Avoiding the allocation can even be an incremental change.

Reply:
What about: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2648/files#diff-0bf3e7afdfb06195ce65362ac431a7b2c8baadb1cab0745231b78f1a5c6500b2R104-R110?